### PR TITLE
api(ticdc): fix panic after middleware refactor

### LINKED ISSUE #5888: Non-owner cdc server panic on cdc open-api requests
### What did you do?

I launched a TiDB cluster and 2 TiCDC servers based on it. Then I send API requests to the CDC server, which is not the owner. That server incurred a panic though it successfully forward the request to the owner.

### What did you expect to see?

No panic

### What did you see instead?

The server panicked.

```log
2022/06/15 13:08:50 [Recovery] 2022/06/15 - 13:08:50 panic recovered:
GET /api/v1/changefeeds HTTP/1.1
Host: 10.2.5.138:8302
Accept: */*
Accept-Encoding: gzip, deflate, br
Connection: keep-alive
User-Agent: apifox/1.0.0 (https://www.apifox.cn)


runtime error: invalid memory address or nil pointer dereference
runtime/panic.go:220 (0x44ecf5)
runtime/signal_unix.go:818 (0x44ecc5)
github.com/pingcap/tiflow/cdc/api/v1/api.go:130 (0x2a38a2d)
github.com/gin-gonic/gin@v1.7.4/context.go:165 (0x2a16931)
github.com/pingcap/tiflow/cdc/api/middleware/middleware.go:59 (0x2a1691d)
github.com/gin-gonic/gin@v1.7.4/context.go:165 (0x2a16116)
github.com/pingcap/tiflow/cdc/api/middleware/middleware.go:33 (0x2a160f9)
github.com/gin-gonic/gin@v1.7.4/context.go:165 (0x28024c1)
github.com/gin-gonic/gin@v1.7.4/recovery.go:99 (0x28024ac)
github.com/gin-gonic/gin@v1.7.4/context.go:165 (0x280133d)
github.com/gin-gonic/gin@v1.7.4/gin.go:489 (0x2800fc5)
github.com/gin-gonic/gin@v1.7.4/gin.go:445 (0x2800b24)
net/http/server.go:2916 (0x7268fa)
net/http/server.go:1966 (0x721836)
runtime/asm_amd64.s:1571 (0x46c420)
```

### Versions of the cluster

Upstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):

```console
6.1.0
```

Upstream TiKV version (execute `tikv-server --version`):

```console
6.1.0
```

TiCDC version (execute `cdc version`):

```console
v6.0.0-master-dirty
```

### PR DESCRIPTION
**Pull request by crelax**
### What problem does this PR solve?

Issue Number: close #5888

### What is changed and how it works?
called abort in the middleware

#### Tests 
 - Manual test
   - send requests to none-ower CDC server and check the console

##### Will it cause performance regression or break compatibility?
No
##### Do you need to update user documentation, design documentation or monitoring documentation?
No
### Release note 

```release-note
`None`
```

### CODE REVIEW

File: cdc/api/util.go [hunk 1]
```diff
@@ -139,3 +143,88 @@ func HandleOwnerScheduleTable(
 		return errors.Trace(err)
 	}
 }
+
+// forWardFromCapture is a header to be set when
+// a request is forwarded from another capture
+const forWardFromCapture = "TiCDC-ForwardFromCapture"
```
Comment (liuzix) [inline cdc/api/util.go:149]:
```suggestion
const forwardFromCapture = "TiCDC-ForwardFromCapture"
```

File: cdc/api/util.go [hunk 2]
```diff
@@ -139,3 +143,88 @@ func HandleOwnerScheduleTable(
...
+
+// ForwardToOwner forward an request to owner
```
Comment (liuzix) [inline cdc/api/util.go:151]:
```suggestion
// ForwardToOwner forwards an request to the owner
```

File: cdc/api/middleware/middleware.go [hunk 3]
```diff
@@ -73,3 +73,16 @@ func ErrorHandleMiddleware() gin.HandlerFunc {
...
 	}
 }
+
+// ForwardToOwnerMiddleware forward an request to owner if current server
+// is not owner, or handle it locally.
+func ForwardToOwnerMiddleware(p api.CaptureInfoProvider) gin.HandlerFunc {
+	return func(ctx *gin.Context) {
+		if !p.IsOwner() {
+			api.ForwardToOwner(ctx, p)
+			ctx.Abort()
```
Comment (liuzix) [inline cdc/api/middleware/middleware.go:83]: Please add an comment on the importance of calling `Abort()`.

Comment (crelax) [reply cdc/api/middleware/middleware.go:87]: Got it

**Review by liuzix — APPROVED** [review]

**Review by sdojjy — APPROVED** [review]

**Pull request merged**